### PR TITLE
Glossary updates for architectural terms

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jenniferfell** [comment]
Tagged Reviewers: Not on the v6 track, but we wanted to start the process. First installment of general glossary updates and recommendations. This installment focuses on terms used in the architecture doc. Thanks!

File: aio/content/guide/glossary.md [hunk 1]
```diff
@@ -459,129 +343,93 @@ Angular calls these hook methods in the following order:
...
 
-To use observables, Angular uses a third-party library called Reactive Extensions (RxJS).
-Observables are a proposed feature for ES2016, the next version of JavaScript.
+## Observer
 
+An object passed to the subscriber function for an [observable](guide/glossary#observable).
+A JavaScript object with up to three callbacks, one for each type of notification that an observable can deliver: 
+- The `next` notification: sends a value such as a Number, a String, an Object, etc.
+- The `error` notification sends a JavaScript Error or exception.
+- The `complete` notification does not send a value, but the handler is called when the call completes. Values can continue to be returned after the call completes
```
Comment (trotyl) [inline aio/content/guide/glossary.md]: `Values can continue to be returned after the call completes`. Really?

File: aio/content/guide/glossary.md [hunk 2]
```diff
@@ -711,24 +636,31 @@ Read about how to write template expressions
...
+
+An opaque identifier used for efficient table lookup. In Angular, a [DI token](guide/glossary#di-token) is used to find [providers](guide/glossary#provider) of dependencies in the [dependency injection](guide/glossary#di) system.
+
+{@a transpile}
 
 ## Transpile
 
-The process of transforming code written in one form of JavaScript
-(such as TypeScript) into another form of JavaScript. (See also [ECMAScript](guide/glossary#ecma)).
+Translate-and-compile is the process by which Angular transforms TypeScript source code to JavaScript for delivery.
```
Comment (IgorMinar) [inline aio/content/guide/glossary.md]: The process of translation of transforming one version of JavaScript code to another version JavaScript, For example down-leveling ES 2015 JavaScript to the older ES5 version.

File: aio/content/guide/glossary.md [hunk 3]
```diff
@@ -711,24 +636,31 @@ Read about how to write template expressions
...
 
-{@a typescript}
+Angular offers [ahead-of-time (AOT) compilation](guide/glossary#aot) for production and [just-in-time (JIT) compilation](guide/glossary#jit) for the development environment.
```
Comment (IgorMinar) [inline aio/content/guide/glossary.md]: this is not right because it has nothing to do with transpilation. I'd delete it.

**Review by IgorMinar — CHANGES_REQUESTED** [review]
this needs some tweaks but most of the changes are quite good! thanks

File: aio/content/guide/glossary.md [hunk 4]
```diff
@@ -202,10 +169,24 @@ operations and supporting declaration syntax.
  * [Style binding](guide/template-syntax#style-binding).
  * [Two-way data binding with ngModel](guide/template-syntax#ngModel).
 
+{@a declarabel}
 
-{@a decorator}
+## Declarable
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: This definition comes from the NgModules guide: https://angular.io/guide/feature-modules#how-to-make-a-feature-module

File: aio/content/guide/glossary.md [hunk 5]
```diff
@@ -668,21 +579,33 @@ For more information, see the [Services](tutorial/toh-pt4) page of the [Tour of
...
-A category of [directive](guide/glossary#directive) that can
-shape or reshape HTML layout, typically by adding and removing elements in the DOM.
-The `ngIf` "conditional element" directive and the `ngFor` "repeater" directive are well-known examples.
+A category of [directive](guide/glossary#directive) that is responsible for shaping or reshaping HTML layout by modifying the DOM (adding, removing, or manipulating elements and their children).
+
+To learn more, see the [Structural Directives](guide/structural-directives) page.
+
+{@a subscriber}
+
+## Subscriber
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: The observer term is used in the doc. Moving the details to Subscriber.

File: aio/content/guide/glossary.md [hunk 6]
```diff
@@ -305,34 +253,25 @@ Read more in the [Dependency Injection](guide/dependency-injection) page.
 
 ## Directive
 
-An Angular class responsible for creating, reshaping, and interacting with HTML elements
-in the browser DOM. The directive is Angular's most fundamental feature.
+A class definition with the `@Directive` [decorator](guide/glossary#decorator) that can modify the structure of the DOM, or modify attributes in the DOM and component data model.
```
[the remark this answers is not in the document]
Comment (trotyl) [reply aio/content/guide/glossary.md]: Technically there're indeed two types of `class` syntax, `ClassDeclaration` and `ClassExpression`, but the current decorator proposal only allows `ClassDeclaration`.

File: aio/content/guide/glossary.md [hunk 7]
```diff
@@ -238,63 +219,30 @@ classes that follow it in the file.
 <div class="alert is-important">
 
 
+In Angular, a function that adds metadata to a class, or to  its members and function arguments. Decorators are sometimes called annotations. Decorators are an experimental (stage 2), JavaScript language [feature](https://github.com/wycats/javascript-decorators). TypeScript adds support for decorators.
```
[the remark this answers is not in the document]
Comment (trotyl) [reply aio/content/guide/glossary.md]: The URL provided is not the [stage-2 decorator](https://github.com/tc39/proposal-decorators), but the original stage-0 community one.

File: aio/content/guide/glossary.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: Just using "class" per Igor's suggestion

File: aio/content/guide/glossary.md [hunk 8]
```diff
@@ -454,118 +410,95 @@ Angular calls these hook methods in the following order:
...
+{@a observable}
 
-Every Angular application has an application root-module class. By convention, the class is
-called `AppModule` and resides in a file named `app.module.ts`.
 
-For details and examples, see [NgModules](guide/ngmodules) and the 
-related files in that section.
+## Observable
 
+A producer of multiple values, which it pushes to [subscribers](guide/glossary#subscriber). Used for asynchronous event handling throughout Angular. You execute an observable by subscribing to it, with an [observer](guide/glossary#observer) that contains callbacks for notifications of new values, errors, or completion.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: The observer object doesn't produce values, it defines the handlers. The Observable instance contains the definition of how to product the values.

File: aio/content/guide/glossary.md [hunk 9]
```diff
@@ -454,118 +410,95 @@ Angular calls these hook methods in the following order:
...
-Observables are used within Angular itself, including Angular's event system and its HTTP client service.
+To learn more, see the [Observables](guide/glossary#observable) guide.
 
-Observables are a proposed feature for ES2016, the next version of JavaScript.
-Currently, Angular depends on a third-party library called Reactive Extensions (RxJS) to provide observables.
 
-For more information, see the [Observables guide](guide/observables).
+{@a observer}
 
+## Observer
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: We use it in our published doc and it is well defined there.
 @benlesh -- is this not the correct terminology? If not, we need to revise our doc.

**Review by IgorMinar — APPROVED** [review]
lgtm. I squashed all the commits into one and force-pushed into this branch to get it ready for merging.

**Comment by shaungrady** [comment]
For my own edification, why was `barrel` removed from the glossary? Is this no longer a recommended pattern?

**Pull request closed without merging**